### ISSUE TITLE
"Cannot identify image file"

### ISSUE THREAD

**Issue by dustystu** [deeppomf/DeepCreamPy#170]
Window shows the following text:

2019-09-20 00:02:22.102925: | tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX
Decensor ing the image ./decensor_input/leg.png
Cannot identify image file (./decensor_input/leg.png)
Decensoring complete!

After window closes there is no decensored image in /.../decensor_output/

**To Reproduce**
1. Go to /.../deepPy/
2. follow instructions in usage.md, color censored image green with Gimp.
using colored image 'leg.png' in /.../decensor_input/ and original image 'leg.png' in /.../decensor_input_original/
3. Click on decensor.exe

**Expected behavior**
decensored image

**Installation method**
Did you install DeepCreamPy with the binary? no? I extracted it from DeepCreamPy_2.1.0-beta_win64.zip
Did you run DeepCreamPy's code yourself? used decensor.exe

**Desktop (please complete the following information):**
 - OS: win 10 pro 64bit
 -  Version 10.0.17763 Build 17763 or whatever is most recent
![words](https://user-images.githubusercontent.com/55576222/65307618-03844980-db3d-11e9-8c27-4578afb9f159.png)
[msinfotxt.txt](https://github.com/deeppomf/DeepCreamPy/files/3634666/msinfotxt.txt)

 - Supports AVX? idk

**Additional context**

**Comment by deeppomf** [comment]
Can you send me the image you want to decensor through twitter @deeppomf  or reddit /u/deeppomf?

**Comment by RuhtraX** [comment]
I'm taking a guess that it's color indexing. Did you check in GIMP if the image uses indexed colors?

**Comment by dustystu** [comment]
I tried several different .png's. They all gave the above error message ("Cannot identify image").
I used GIMP for all file edits. 
I tried with pictures that started as .pngs and I also tried with image that I converted to .pngs with gimp -> export as .png
Is there some setting in GIMP I have to double check? Do I need to figure out how to use 'index colors' mode?

**Comment by dustystu** [comment]
Update:
I finally got it to work. 
I used a 'raw' .png (not converted from other extension)
I colored the censor bar green with MSPaint
run decensor.exe
-> decensored image as expected.

**Comment by dustystu** [comment]
So I guess my new question is this:
How Do I use Gimp to color green + export as .png and have it work correctly?

**Comment by ccppoo** [comment]
PIL used in DeepCreamPy demand 8bpc RGB or 8bpc RGBA (RBG 0~255), it seems your image could be saved as 16bpc I guess? 

related issue on python PIL : https://github.com/python-pillow/Pillow/issues/2955
Could you send me the image or post it on r/DeepCreamPy to debug?

**Comment by dustystu** [comment]
@ccppoo 
I'm not sure why giving you the image matters because it's any image I export with GIMP will give this error. 
Trigger warnings: nudity, pregnancy, sex
[8a4e1c6c3b2463e70c6c985e7e357b4e.zip](https://github.com/deeppomf/DeepCreamPy/files/3638457/8a4e1c6c3b2463e70c6c985e7e357b4e.zip)

In any case, I tried exporting with the 8bit rgb and 8bit rgba, and I got the same error.

**Comment by ccppoo** [comment]
@dustystu
I'll check it

**Comment by ccppoo** [comment]
@dustystu 
When I checked with java :
![image](https://user-images.githubusercontent.com/46418312/65374994-2c3a4980-dccb-11e9-9efc-567ea1fedc56.png)

When I used pngcheck : 
```txt
m.png  CRC error in chunk iCCP (computed b1f9dfd7, expected 88fedf6f)
ERROR: m.png
```
[Wiki crc error](https://en.wikipedia.org/wiki/Cyclic_redundancy_check)

as pngcheck is GPL, I can't implement this right away

**Comment by dustystu** [comment]
@ccppoo 
So tldr; deepcreampy is NOT compatible with the encoding GIMP uses to export .pngs? I should use mspaint for now? (I don't have photoshop).

**Comment by ccppoo** [comment]
@dustystu
No, it doen't matter which software you use. What do matter is format.
+ DeepCreamPy(PIL to be tell exactly) takes(adjusts) image as 8-bit(24bit or 32bit per pixel).
so if you want to make decensored image more high definition, use waifu2

**Comment by RuhtraX** [comment]
I got the same error with your png file, I fixed it by disabling all options like "save resolution" "save thumbnail" etc. and exporting as 8bit RGB and it worked

![Gimp_Screenshot](https://user-images.githubusercontent.com/12171146/65392235-ecb24280-dd48-11e9-9885-d64506eab352.png)
I guess the last option ("Save color profile") is the culprit, but I haven't tested disabling only that option
